### DISCUSSION TITLE
Keycloak Admin REST API - list client scopes attached to role

### DISCUSSION THREAD

**Discussion by Ansis100** [keycloak/keycloak#21308] [Q&A]
Is there any way to list client scopes that are attached to, for example, a realm role?

I find that
1) It is a bit unintuitive how in the Web UI you *have* to assign roles to scopes and not the other way around;
2) It is impossible to use the REST API to find out what scopes are available in a role.

The only way would be iterating through *ALL* client scopes and checking if there is a mapping for the specific role we're looking for. That's going to take a while.

**Comment by jatinderkumargupta** [comment]
I want to create realm through Admin API. What permissions do we need?

I am not able to add it through API. Please help.
